### PR TITLE
Making the login path nicer

### LINKED ISSUE #609: WebUI - error message on invalid username
The web UI returns the following error when an invalid username is entered in the login flow:
```
An error has occured 😔

invalid type: string "nomatchingentries", 
expected struct AuthResponse at line 1 column 19
```

### PR DESCRIPTION
**Pull request by yaleman**
Fixes #609 

- throws a specific error on user-not-found and account lockout.
- adds a console_log unsafe function because `rustanalyser` was yelling at me.
- adds a script to make it easier to run a dev server instead of having to remember the command every time 😄 

- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

Some shiny new UI for when you try to log in as a locked account:

Screen Shot 2022-04-28 at 19 39 37

Or an invalid one:

Screen Shot 2022-04-28 at 19 39 31

### CODE REVIEW

File: kanidmd/idm/src/valueset.rs [hunk 1]
```diff
@@ -1058,17 +1058,11 @@ impl ValueSet {
             I::Utf8(set) => set.iter().cloned().collect(),
             I::Iutf8(set) => set.iter().cloned().collect(),
             I::Iname(set) => set.iter().cloned().collect(),
-            I::Uuid(set) => set
-                .iter()
-                .map(|u| u.as_hyphenated().to_string())
-                .collect(),
+            I::Uuid(set) => set.iter().map(|u| u.as_hyphenated().to_string()).collect(),
```
Comment (Firstyear) [inline kanidmd/idm/src/valueset.rs:1061]: What version of rust are you using? This kind of formatting back-and-forth I've seen before when there are different versions ....

File: kanidmd_web_ui/src/login.rs [hunk 2]
```diff
@@ -235,7 +251,9 @@ impl LoginApp {
                     <div class="container">
                         <form
                             onsubmit={ ctx.link().callback(|e: FocusEvent| {
-                                console::log!("login::view_state -> Password - prevent_default()");
+                                unsafe {
```
Comment (Firstyear) [inline kanidmd_web_ui/src/login.rs:254]: Why is this unsafe needed? Why change from console::log! ?

File: kanidmd_web_ui/src/login.rs [hunk 3]
```diff
@@ -405,13 +455,19 @@ impl LoginApp {
     }
 }
 
+unsafe fn console_log(message: String) {
```
Comment (Firstyear) [inline kanidmd_web_ui/src/login.rs:458]: Why does this need to be unsafe?

**Review by Firstyear — CHANGES_REQUESTED** [review]
I don't think the unsafe on console is needed?

File: kanidmd/idm/src/valueset.rs (hunk 1 above)
Comment (yaleman) [reply kanidmd/idm/src/valueset.rs:1061]: Oh I'm running 1.60, I'll push back to 1.59 and see what it shows.

Comment (Firstyear) [reply kanidmd/idm/src/valueset.rs:1061]: I'm on 1.60 too, so that's extra haunted.

File: kanidmd_web_ui/src/login.rs (hunk 3 above)
Comment (yaleman) [reply kanidmd_web_ui/src/login.rs:458]: `rust-analyzer` had a hissy-fit about `gloo::console::log!()` being unsafe, I will revert the changes if it's invalid 😇

Comment (Firstyear) [reply kanidmd_web_ui/src/login.rs:458]: Yeah, rust-analyzer is being silly. It's fine.

Comment (Firstyear) [reply kanidmd_web_ui/src/login.rs:458]: Yeah, it's being really silly, there is no unsafe in gloo console at all, so I have nfi what's going on here. Perhaps it's somewhere in the wasm-bindgen bits underneath, but this is NOT unsafe at all from what I can see.

Comment (yaleman) [reply kanidmd_web_ui/src/login.rs:458]: I'll revert, won't take long to do 🥳

Comment (Firstyear) [reply kanidmd_web_ui/src/login.rs:458]: Thanks :)

**Review by Firstyear — APPROVED** [review]

File: kanidmd/idm/src/valueset.rs (hunk 1 above)
Comment (yaleman) [reply kanidmd/idm/src/valueset.rs:1061]: I went back to 1.56 and it didn't want me to change anything so... meh? Leaving this alone for now.

**Pull request merged**